File: backend/app/schemas/cover_letter.py
from pydantic import BaseModel
from typing import List, Optional


class CoverLetterGenerateRequest(BaseModel):
    cv_id: str
    job_description: str
    tone: str = "professional"  # professional, enthusiastic, concise, creative
    format: str = "us"  # us, french, international
    language: str = "en"
    custom_instructions: Optional[str] = None


class CoverLetterRewriteRequest(BaseModel):
    paragraph_index: int
    current_text: str
    tone: str = "professional"
    instructions: Optional[str] = None


class CoverLetterVersion(BaseModel):
    id: str
    tone: str
    created_at: str


class CoverLetterContent(BaseModel):
    id: Optional[str] = None
    cv_id: str
    paragraphs: List[str]
    tone: str
    format: str
    language: str
    word_count: int
    created_at: Optional[str] = None


class CoverLetterDownloadRequest(BaseModel):
    paragraphs: List[str]
    format: str = "pdf"  # pdf, docx
    tone: str = "professional"
    letter_format: str = "us"

File: frontend/lib/api/cover-letter.ts
import { request, download } from "./client";

export type CoverLetterTone = "professional" | "enthusiastic" | "concise" | "creative";
export type CoverLetterFormat = "us" | "french" | "international";

export interface CoverLetterContent {
  id?: string;
  cv_id: string;
  paragraphs: string[];
  tone: CoverLetterTone;
  format: CoverLetterFormat;
  language: string;
  word_count: number;
  created_at?: string;
}

export interface CoverLetterVersion {
  id: string;
  tone: string;
  created_at: string;
}

export const coverLetterApi = {
  generate: (data: {
    cv_id: string;
    job_description: string;
    tone?: CoverLetterTone;
    format?: CoverLetterFormat;
    language?: string;
    custom_instructions?: string;
  }) =>
    request<CoverLetterContent>("/cover-letter/generate", {
      method: "POST",
      body: JSON.stringify(data),
    }),

  rewriteParagraph: (paragraphIndex: number, currentText: string, tone = "professional", instructions?: string) =>
    request<{ rewritten_text: string }>("/cover-letter/rewrite-paragraph", {
      method: "POST",
      body: JSON.stringify({
        paragraph_index: paragraphIndex,
        current_text: currentText,
        tone,
        instructions,
      }),
    }),

  saveVersion: (clId: string) =>
    request<CoverLetterVersion>(`/cover-letter/${clId}/save-version`, {
      method: "POST",
    }),

  listVersions: (clId: string) =>
    request<CoverLetterVersion[]>(`/cover-letter/${clId}/versions`),

  download: (paragraphs: string[], format: "pdf" | "docx" = "pdf", tone = "professional", letterFormat = "us") =>
    download("/cover-letter/download", {
      method: "POST",
      body: JSON.stringify({ paragraphs, format, tone, letter_format: letterFormat }),
    }),
};
